src/chain/mod.rs

```diff
@@ -1514,84 +1537,87 @@ impl ChainSource {
 	}
 }
 
-impl ChainSource {
+impl EsploraChainSource {
 	pub(crate) async fn process_broadcast_queue(&self) {
-		match &self.kind {
-			ChainSourceKind::Esplora { esplora_client, tx_broadcaster, logger, .. } => {
-				let mut receiver = tx_broadcaster.get_broadcast_queue().await;
-				while let Some(next_package) = receiver.recv().await {
-					for tx in &next_package {
-						let txid = tx.compute_txid();
-						let timeout_fut = tokio::time::timeout(
-							Duration::from_secs(TX_BROADCAST_TIMEOUT_SECS),
-							esplora_client.broadcast(tx),
-						);
-						match timeout_fut.await {
-							Ok(res) => match res {
-								Ok(()) => {
+		let mut receiver = self.tx_broadcaster.get_broadcast_queue().await;
+		while let Some(next_package) = receiver.recv().await {
+			for tx in &next_package {
+				let txid = tx.compute_txid();
+				let timeout_fut = tokio::time::timeout(
+					Duration::from_secs(TX_BROADCAST_TIMEOUT_SECS),
+					self.esplora_client.broadcast(tx),
+				);
+				match timeout_fut.await {
+					Ok(res) => match res {
+						Ok(()) => {
+							log_trace!(self.logger, "Successfully broadcast transaction {}", txid);
+						},
+						Err(e) => match e {
+							esplora_client::Error::HttpResponse { status, message } => {
+								if status == 400 {
+									// Log 400 at lesser level, as this often just means bitcoind already knows the
+									// transaction.
+									// FIXME: We can further differentiate here based on the error
+									// message which will be available with rust-esplora-client 0.7 and
+									// later.
 									log_trace!(
-										logger,
-										"Successfully broadcast transaction {}",
-										txid
+										self.logger,
```

Comment:
Glad you did this somewhat artificial commit struct. Very easy to ascertain that it's still the same.